tests/Builder/PipelineTestCase.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MongoDB\Tests\Builder;
+
+use MongoDB\Builder\BuilderEncoder;
+use MongoDB\Builder\Pipeline;
+use PHPUnit\Framework\TestCase;
+
+use function MongoDB\BSON\fromJSON;
+use function MongoDB\BSON\toPHP;
+use function var_export;
+
+class PipelineTestCase extends TestCase
+{
+    final public static function assertSamePipeline(string $expected, Pipeline $pipeline): void
```

Comment:
I assume this is the first use of this function. If we keep it as-is, renaming `$expected` to `$expectedJson` may be clearer. I do think it may be preferable to avoid JSON entirely and just generate PHP files that return a value that can be included here.

Those generated files could then have the appropriate "auto-generated" header and be ignored via `.gitattributes`.